### ISSUE TITLE
Fix alpha blending for X-ray

### ISSUE THREAD

**Issue by tothatt81** [xeokit/xeokit-sdk#318]
Hey Lindsay,

We have found some problem with the xray material settings, it even affects the presets. I put together a test file, so you can reproduce the issue:
https://gist.github.com/tothatt81/015cda11bb22e16b98be7aae432bdd8a

What I found is that it looks good when either the fillAlpha or the edgeAlpha property is 0 or 1, but when both are fractions then it looks like this:
![xray](https://user-images.githubusercontent.com/39777981/83043011-979a9f00-a042-11ea-8fce-c2e91827ed08.PNG)

Would you please look into it? Thanks!

**Comment by xeolabs** [comment]
@tothatt81 is the issue that the fill colors are "washing out" or "whiting out" too much, as shown at the center of the screenshot you've posted?

**Comment by tothatt81** [comment]
Hey Lindsay,

I have just checked the latest commit (0097b0a) and for me it still doesn't look right.

Just for reference we use the defaultWhiteBG preset with edgeAlpha 0.8 and fillAlpha 0.15. On version 0.9.1 it looked like this:
![xray_old](https://user-images.githubusercontent.com/39777981/83560332-5872bf00-a516-11ea-83b1-4f9384979b39.PNG)

This is what we would like to achieve with the current version as well, but the latest one with same settings shows this view:
![xray_new](https://user-images.githubusercontent.com/39777981/83560352-61639080-a516-11ea-9169-41d79131258e.PNG)

**Comment by xeolabs** [comment]
Hey @tothatt81 I just dug further and found one more part to this issue, which I've fixed in 145ca17, see if that works for you.

\ping @tmarti - this might affect you.

I found that the issue is not just the blending function but the use of ````gl.depthMask(true)```` when rendering transparent objects.

````gl.depthMask```` sets whether writing into the depth buffer is enabled or disabled when rendering transparent fill faces and edges.

When ````true````, transparent objects are rendered without z-buffer testing, which means that
 all transparent objects will be visible, however may color each other in unexpected ways from
 different angles.

 When ````false````, transparent objects are rendered with z-buffer testing, which means that you lose some of your interior surfaces because z-buffering will cause some nearer transparent surfaces  to replace some farther surfaces when the nearer surfaces were rendered first.
    
````gl.depthMask```` was set ````false```` by default.

I exposed it as a new configuration for ````Viewer````, which by default sets it ````true````.

In case you need to disable it, instantiate your ````Viewer````  like so:

````javascript
const myViewer = new Viewer({
    alphaDepthMask: false
});
````

**Comment by tothatt81** [comment]
Looks good to me now, thank you.
